Spigot-Server-Patches/0402-Fix-NPE-when-setting-a-player-s-spectator-target.patch

```diff
@@ -0,0 +1,215 @@
+ import org.bukkit.event.player.PlayerLocaleChangeEvent;
+ import org.bukkit.event.player.PlayerTeleportEvent.TeleportCause;
+ import org.bukkit.inventory.MainHand;
++
++import javax.annotation.Nullable;
++import java.util.*;
++
++// CraftBukkit start
+ // CraftBukkit end
+ 
+ public class EntityPlayer extends EntityHuman implements ICrafting {
+@@ -62,9 +54,19 @@ public class EntityPlayer extends EntityHuman implements ICrafting {
+     private EntityHuman.EnumChatVisibility cs;
+     private boolean ct = true;
+     private long cu = SystemUtils.b();
+-    private Entity cv; private void setSpectatorTargetField(Entity e) { this.cv = e; } // Paper - OBFHELPER
++    private Entity cv;
++
++    private void setSpectatorTargetField(Entity e) {
++        this.cv = e;
++    } // Paper - OBFHELPER
++
+     public boolean worldChangeInvuln;
+-    private boolean cx; private void setHasSeenCredits(boolean has) { this.cx = has; } // Paper - OBFHELPER
++    private boolean cx;
++
++    private void setHasSeenCredits(boolean has) {
++        this.cx = has;
++    } // Paper - OBFHELPER
++
+     private final RecipeBookServer cy;
+     private Vec3D cz;
+     private int cA;
+@@ -535,8 +537,8 @@ public class EntityPlayer extends EntityHuman implements ICrafting {
+             this.playerConnection.a((Packet) (new PacketPlayOutCombatEvent(this.getCombatTracker(), PacketPlayOutCombatEvent.EnumCombatEventType.ENTITY_DIED, ichatbasecomponent)), (future) -> {
+                 if (!future.isSuccess()) {
+                     String s = ichatbasecomponent.a(256);
+-                    ChatMessage chatmessage = new ChatMessage("death.attack.message_too_long", new Object[] { (new ChatComponentText(s)).a(EnumChatFormat.YELLOW)});
+-                    IChatBaseComponent ichatbasecomponent1 = (new ChatMessage("death.attack.even_more_magic", new Object[] { this.getScoreboardDisplayName()})).a((chatmodifier) -> {
++                    ChatMessage chatmessage = new ChatMessage("death.attack.message_too_long", new Object[]{(new ChatComponentText(s)).a(EnumChatFormat.YELLOW)});
++                    IChatBaseComponent ichatbasecomponent1 = (new ChatMessage("death.attack.even_more_magic", new Object[]{this.getScoreboardDisplayName()})).a((chatmodifier) -> {
+                         chatmodifier.setChatHoverable(new ChatHoverable(ChatHoverable.EnumHoverAction.SHOW_TEXT, chatmessage));
+                     });
+ 
+@@ -687,7 +689,8 @@ public class EntityPlayer extends EntityHuman implements ICrafting {
+             this.world.kill(this);
+             if (!this.viewingCredits) {
+                 this.viewingCredits = true;
+-                if (world.paperConfig.disableEndCredits) this.setHasSeenCredits(true); // Paper - Toggle to always disable end credits
++                if (world.paperConfig.disableEndCredits)
++                    this.setHasSeenCredits(true); // Paper - Toggle to always disable end credits
```

Comment:
Same thing here

Reply:
Yea I am already fixing it right now sorry. IntelliJ formatted it and I didn't realize.